src/client/interpreter/configuration/interpreterSelector/commands/setInterpreter.ts

```diff
@@ -166,4 +182,32 @@ export class SetInterpreterCommand extends BaseInterpreterSelectorCommand {
             await this.pythonPathUpdaterService.updatePythonPath(interpreterState.path, configTarget, 'ui', wkspace);
         }
     }
+
+    /**
+     * Check if the interpreter that was entered exists in the cache.
+     * If it does, it means that it had already been discovered,
+     * and we didn't do a good job of surfacing it.
+     *
+     * @param selection Intepreter path that was either entered manually or picked by browsing through the filesystem.
+     */
+    private async sendInterpreterEntryTelemetry({ path: selection, workspace }: InterpreterStateArgs): Promise<void> {
+        let interpreterPath = path.normalize(untildify(selection));
+
+        if (!path.isAbsolute(interpreterPath)) {
+            interpreterPath = path.resolve(workspace?.fsPath || '', selection!);
+        }
+
+        let discovered: boolean;
+        if (await inDiscoveryExperiment(this.experiments)) {
+            const env = await this.pyenvs.getInterpreterCache(interpreterPath, this.context);
+            discovered = !!env;
+        } else {
+            const store = await this.interpreterService.getInterpreterCache(interpreterPath);
+            discovered = store.value && store.value.info !== undefined;
+        }

```

Comment:
As noted in my other comment, we should preserve the list of suggested envs from `_pickInterpreter()` and check it, rather than making the comparison from any other source.

So this would be more like:

```suggestion
        const discovered = suggestions.map((s) => s.path).includes(interpreterPath);
```